unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/QuestsTrackerHUD/QuestsTrackerEntry.cs

```diff
@@ -0,0 +1,131 @@
+using System.Linq;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Huds.QuestsTracker
+{
+    public class QuestsTrackerEntry : MonoBehaviour
+    {
+        public event Action OnLayoutRebuildRequested;
+
+        [SerializeField] internal TextMeshProUGUI questTitle;
+        [SerializeField] internal RawImage questIcon;
+        [SerializeField] internal TextMeshProUGUI sectionTitle;
+        [SerializeField] internal Image progress;
+        [SerializeField] internal RectTransform tasksContainer;
+        [SerializeField] internal GameObject taskPrefab;
+        [SerializeField] internal Button expandCollapseButton;
+        [SerializeField] internal GameObject expandIcon;
+        [SerializeField] internal GameObject collapseIcon;
+        [SerializeField] internal Toggle pinQuestToggle;
+        [SerializeField] internal RawImage iconImage;
+
+        private AssetPromise_Texture iconPromise;
+
+        internal QuestModel quest;
+        private bool isExpanded;
+        private static BaseCollection<string> pinnedQuests => DataStore.i.Quests.pinnedQuests;
+
+        public void Awake()
+        {
+            pinQuestToggle.onValueChanged.AddListener(OnPinToggleValueChanged);
+
+            expandCollapseButton.gameObject.SetActive(false);
+            SetExpandCollapseState(true);
+            expandCollapseButton.onClick.AddListener(() => SetExpandCollapseState(!isExpanded));
+        }
+
+        public void Populate(QuestModel newQuest)
+        {
+            quest = newQuest;
+            questTitle.text = quest.name;
+            SetIcon(quest.icon);
+            QuestSection currentSection = quest.sections.First(x => x.progress < 1f);
+            sectionTitle.text = $"{currentSection.name} - {currentSection.progress * 100}%";
+            progress.fillAmount = currentSection.progress;
+
+            CleanUpTasksList();
+            foreach (QuestTask task in currentSection.tasks)
+            {
+                CreateTask(task);
+            }
+            expandCollapseButton.gameObject.SetActive(currentSection.tasks.Length > 0);
+            SetExpandCollapseState(true);
+        }
+
+        internal void CreateTask(QuestTask task)
+        {
+            var taskUIEntry = Instantiate(taskPrefab, tasksContainer).GetComponent<QuestsTrackerTask>();
+            taskUIEntry.Populate(task);
+        }
+
+        internal void SetIcon(string iconURL)
+        {
+            if (iconPromise != null)
+            {
+                iconPromise.ClearEvents();
+                AssetPromiseKeeper_Texture.i.Forget(iconPromise);
+            }
+
+            if (string.IsNullOrEmpty(iconURL))
+                return;
+
+            iconPromise = new AssetPromise_Texture(iconURL);
+            iconPromise.OnSuccessEvent += OnIconReady;
+            iconPromise.OnFailEvent += x => { Debug.Log($"Error downloading quest tracker entry icon: {iconURL}"); };
+
+            AssetPromiseKeeper_Texture.i.Keep(iconPromise);
```

Comment:
Caution: this promise isn't being forget when this entry is destroyed. The texture will dangle in memory.

Reply:
Nice catch! in fact there are a couple more of promises not being forgotten. Gonna fix it!